frontend/src/components/WorkoutPlan/WorkoutPlan.tsx

```diff
@@ -82,17 +85,24 @@ const WorkoutPlan: FC<WorkoutPlanProps> = () => {
     await setItem(JSON.stringify(session));
   };
 
-  useEffect(() => {
+  const getWorkoutPlan = () => {
```

Comment:
We should put this in a react query. Make sure to keep the functionality of when the data is retrieved though